idp/usso/ussooauth/ussooauth.go

```
@@ -133,7 +133,10 @@ var consumerKeyRegexp = regexp.MustCompile(`oauth_consumer_key="([^"]*)"`)
 // verifyOAuthSignature verifies with Ubuntu SSO that the request is correctly
 // signed.
 func verifyOAuthSignature(requestURL string, req *http.Request) (string, error) {
-	req.ParseForm()
+	err := req.ParseForm()
+	if err != nil {
+		return "", errgo.Notef(err, "cannot parse request form")
```

Fixes a bug where form parsing errors were not handled.